spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/metrics/MetricsEndpointTests.java

```diff
@@ -88,10 +87,22 @@ public void metricValuesAreTheSumOfAllTimeSeriesMatchingTags() {
 		assertThat(getCount(response)).hasValue(4.0);
 	}
 
+	@Test
+	public void metricTagValuesAreDeduplicated() {
+		this.registry.counter("cache", "host", "1", "region", "east", "result", "hit");
+		this.registry.counter("cache", "host", "1", "region", "east", "result", "miss");
+		MetricsEndpoint.MetricResponse response = this.endpoint.metric("cache",
+				Collections.singletonList("host:1"));
+		assertThat(response.getAvailableTags()
+				.stream()
+				.filter(t -> t.getTag().equals("region"))
+				.flatMap(t -> t.getValues().stream()))
+				.containsExactly("east");
+	}
+
 	@Test
 	public void metricWithSpaceInTagValue() {
 		this.registry.counter("counter", "key", "a space").increment(2);
-		MockClock.clock(this.registry).add(SimpleConfig.DEFAULT_STEP);
```

Comment:
Is that related?

Reply:
This line is unnecessary as of rc5

Reply:
Thanks but as it is not related to the purpose of this PR, I've polished this separately in https://github.com/spring-projects/spring-boot/commit/1962cbb6234dd9029bf6a4bb6374d793ba5de959